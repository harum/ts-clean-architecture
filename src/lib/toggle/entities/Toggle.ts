import ToggleAble from './ToggleAble';

export default class Toggle implements ToggleAble {
  #active: boolean;

  constructor(active = false) {
    this.#active = active;
  }

  get active(): boolean {
    return this.#active;
  }

  toggle(): void {
    this.#active = !this.#active;
  }

  on(): void {
    this.#active = true;
  }

  off(): void {
    this.#active = false;
  }
}
