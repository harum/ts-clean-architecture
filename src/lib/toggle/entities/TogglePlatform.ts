import Toggle from './Toggle';

export default class TogglePlatform extends Toggle {
  platform: string;

  constructor(platform = '') {
    super(false);
    this.platform = platform;
  }

  buildFromHash(platformHash: { platform: string; active: boolean }) {
    this.platform = platformHash.platform;
    if (platformHash.active) {
      this.on();
    } else {
      this.off();
    }
  }

  toString(): string {
    return JSON.stringify(this.toHash());
  }

  toHash(): { platform: string; active: boolean } {
    return {
      platform: this.platform,
      active: this.active,
    };
  }
}
