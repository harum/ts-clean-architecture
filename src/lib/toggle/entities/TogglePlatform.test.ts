import Toggle from './Toggle';
import TogglePlatform from './TogglePlatform';

describe('TogglePlatform', () => {
  describe('When initialize the class', () => {
    it('should be a Toggle', () => {
      const toggle = new TogglePlatform('mobile_web');
      toggle.platform = 'mobile_web';
      expect(toggle instanceof Toggle).toBe(true);
    });

    it('should be able to specify platform', () => {
      const toggle = new TogglePlatform('mobile_web');
      expect(toggle.platform).toBe('mobile_web');
    });
  });

  describe('When buildFromHash method is called', () => {
    it('should be able to create toggle platform from hash', () => {
      const toggle = new TogglePlatform();
      toggle.buildFromHash({ platform: 'android', active: true });
      expect(toggle.platform).toBe('android');
      expect(toggle.active).toBe(true);
    });
  });

  describe('When toString method is called', () => {
    it('should return toggle platform info', () => {
      const toggle = new TogglePlatform('mobile_web');
      expect(toggle.toString()).toMatchSnapshot();
    });
  });

  describe('When toHash method is called', () => {
    it('should return toggle platform as hash', () => {
      const toggle = new TogglePlatform('mobile_web');
      expect(toggle.toHash()).toMatchSnapshot();
    });
  });
});
