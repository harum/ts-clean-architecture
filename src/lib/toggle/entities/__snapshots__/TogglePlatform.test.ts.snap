// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TogglePlatform When toHash method is called should return toggle platform as hash 1`] = `
Object {
  "active": false,
  "platform": "mobile_web",
}
`;

exports[`TogglePlatform When toString method is called should return toggle platform info 1`] = `"{\\"platform\\":\\"mobile_web\\",\\"active\\":false}"`;
